base/headers.go

```diff
@@ -161,15 +173,13 @@ func (uri *SipUri) String() string {
 	}
 
 	// Optional userinfo part.
-	if uri.User != nil {
-		buffer.WriteString(*uri.User)
-
-		if uri.Password != nil {
-			buffer.WriteString(":")
-			buffer.WriteString(*uri.Password)
+	switch user := uri.User.(type) {
+	case String:
+		buffer.WriteString(user.String())
+		switch pw := uri.Password.(type) {
+		case String:
+			buffer.WriteString(pw.String())
```

Comment:
Sure, in-fact, have a bunch of tests for the String() method on our types.
I haven't done them all, but it's a start.

Whilst doing that I caught that I accidentally removed the "@" between username and hostname.  That's pretty bad! =P